### PR TITLE
Add registration views

### PR DESCRIPTION
**Pull request by DIdiHL**

### CODE REVIEW

**Comment by DIdiHL** [comment]
Depends on #140

File: app/assets/stylesheets/course/enrol_requests.css.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
+$link-color: #0000ff;
+
+#enrol-request-form {
```
Comment (lowjoel) [inline app/assets/stylesheets/course/enrol_requests.css.scss]: group your selectors according to the controller/action paradigm that winstonyw/rails_utils@1752d6eac348681fb00720538e9b8cb784f935ab uses

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -67,4 +67,18 @@ def link_to_user(user, options = {})
   def format_datetime(date, format = :long)
     date.to_formatted_s(format)
   end
+
+  # Get the course user of the signed in user.
+  #
+  # @return [CourseUser] the course_user instance of the signed in user if a
+  #         course instance variable is defined.
+  def current_course_user
```
Comment (lowjoel) [inline app/helpers/application_helper.rb]: this belongs in the course controller.

File: app/views/course/enrol_requests/_enrol_form.html.slim [hunk 3]
```diff
@@ -0,0 +1,7 @@
+div#enrol-request-form
+  a.btn.btn-primary href="#{new_course_enrol_request_path(course, role: :student)}"
+    | #{t('.enrol_as_student')}
```
Comment (lowjoel) [inline app/views/course/enrol_requests/_enrol_form.html.slim]: `= t(...)`

File: app/views/course/enrol_requests/_enrol_tab.html.slim [hunk 4]
```diff
@@ -0,0 +1,15 @@
+ul class="nav nav-tabs"
+  - if mass_requests
```
Comment (lowjoel) [inline app/views/course/enrol_requests/_enrol_tab.html.slim]: make this output the correct class depending on the active controller instead.

File: app/views/course/enrol_requests/_list.html.slim [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    = no_request_message
+  -else
+    form action=course_process_multiple_enrol_requests_path(@course)
+      table class="table enrol-requests"
+        thead
+        tr
+          - if can? :approve, Course::EnrolRequest
+            td
+
+          td = t('.name')
```
Comment (lowjoel) [inline app/views/course/enrol_requests/_list.html.slim]: `th`.

**Comment by lowjoel** [comment]
I think the views can be refactored. it doesn't feel ruby-ish enough. I've given some comments, we'll iterate.

File: app/assets/stylesheets/course/registrations.css.scss [hunk 6]
```diff
@@ -0,0 +1,37 @@
+$link-color: #0000ff;
```
Comment (lowjoel) [inline app/assets/stylesheets/course/registrations.css.scss]: why is this a separate colour?

File: app/assets/stylesheets/course/registrations.css.scss [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+.course-registrations {
+  &.index {
+    .bold-text {
```
Comment (lowjoel) [inline app/assets/stylesheets/course/registrations.css.scss]: ...just use `strong`?

File: app/assets/stylesheets/course/registrations.css.scss [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      .request-approve,
+      .request-delete {
+        padding-bottom: 2px;
+        padding-top: 2px;
+      }
+
+    }
+
+    .link-style {
+      // Remove all decorations to look like link
```
Comment (lowjoel) [inline app/assets/stylesheets/course/registrations.css.scss]: what's this for?

**Comment by lowjoel** [comment]
I'll come back to this after #140 and #142.

**Comment by lowjoel** [comment]
Merge with #142.

**Pull request closed without merging**